aten/src/ATen/SparseCsrTensorImpl.cpp

```diff
@@ -150,14 +150,22 @@ void SparseCsrTensorImpl::resize_as_sparse_csr_tensor_(const Tensor& src) {
       !has_symbolic_sizes_strides_,
       "resize_as_sparse_csr_tensor_ called on tensor with symbolic shape");
   set_layout(src.layout());
+  Tensor compressed_indices;
+  Tensor plain_indices;
+  std::tie(compressed_indices, plain_indices) =
+      sparse_csr::getCompressedPlainIndices(src);
+
+  // compressed indices are always crow_indices_ on the impl
   crow_indices_ = at::empty_like(
-      src.crow_indices(),
-      src.crow_indices().options(),
-      src.crow_indices().suggest_memory_format());
+      compressed_indices,
+      compressed_indices.options(),
+      compressed_indices.suggest_memory_format());
+  // plain indices are always col_indices_ on the impl
   col_indices_ = at::empty_like(
```

Comment:
To ensure the validity of sparse compressed tensor invariants, use:
```suggestion
  col_indices_ = plain_indices.clone();
```